test/backup.ts

```diff
@@ -0,0 +1,516 @@
+        assert.fail('Should have rethrown error');
+      } catch (thrown) {
+        assert.deepStrictEqual(thrown, err);
+      }
+    });
+  });
+
+  describe('updateExpireTime', () => {
+    const NEW_EXPIRE_TIME = new PreciseDate(1977, 1, 10);
+
+    it('should make the correct request', async () => {
+      const QUERY = {};
+      const ORIGINAL_QUERY = extend({}, QUERY);
+      const expectedReqOpts = extend({}, QUERY, {
+        backup: {
+          name: BACKUP_FORMATTED_NAME,
+          expireTime: NEW_EXPIRE_TIME.toStruct(),
+        },
+        updateMask: {
+          paths: ['expire_time'],
+        },
+      });
+
+      backup.request = config => {
+        assert.strictEqual(config.client, 'DatabaseAdminClient');
+        assert.strictEqual(config.method, 'updateBackup');
+        assert.deepStrictEqual(config.reqOpts, expectedReqOpts);
+
+        assert.notStrictEqual(config.reqOpts, QUERY);
+        assert.deepStrictEqual(QUERY, ORIGINAL_QUERY);
+      };
+
+      await backup.updateExpireTime(NEW_EXPIRE_TIME);
+    });
+
+    it('should accept gaxOpts and a callback', async () => {
+      const options = {
+        timeout: 1000,
+      };
+
+      backup.request = config => {
+        assert.deepStrictEqual(config.gaxOpts, options);
+      };
+
+      await backup.updateExpireTime(NEW_EXPIRE_TIME, options, assert.ifError);
+    });
+
+    it('should return backup object with updated expire time', () => {
+      const API_RESPONSE = {
+        name: 'backup-name',
+        database: 'database-name',
+        expireTime: NEW_EXPIRE_TIME,
+      };
+
+      backup.request = (config, callback: Function) => {
+        callback(null, API_RESPONSE);
+      };
+
+      backup.updateExpireTime(NEW_EXPIRE_TIME, (...args) => {
+        assert.ifError(args[0]);
+        assert.strictEqual(args[1]!.expireTime, NEW_EXPIRE_TIME);
+      });
+    });
+  });
+
+  describe('delete', () => {
+    it('should make the correct request', async () => {
+      const QUERY = {};
+      const ORIGINAL_QUERY = extend({}, QUERY);
+      const expectedReqOpts = extend({}, QUERY, {
+        name: BACKUP_FORMATTED_NAME,
+      });
+
+      backup.request = config => {
+        assert.strictEqual(config.client, 'DatabaseAdminClient');
+        assert.strictEqual(config.method, 'deleteBackup');
+        assert.deepStrictEqual(config.reqOpts, expectedReqOpts);
+
+        assert.notStrictEqual(config.reqOpts, QUERY);
+        assert.deepStrictEqual(QUERY, ORIGINAL_QUERY);
+      };
+
+      await backup.delete();
+    });
+
+    it('should accept gaxOpts and a callback', async () => {
+      const options = {
+        timeout: 1000,
+      };
+
+      backup.request = config => {
+        assert.deepStrictEqual(config.gaxOpts, options);
+      };
+
+      await backup.delete(options, assert.ifError);
+    });
+
+    it('should execute callback with original arguments', done => {
+      const REQUEST_RESPONSE_ARGS = [new Error('Error.'), null];
```

Comment:
I guess since gax only returns the "err" response to use in src/backup.ts, we should mock accordingly:

```diff
- const REQUEST_RESPONSE_ARGS = [new Error('Error.'), null];
+ const REQUEST_RESPONSE_ARGS = [new Error('Error.')];
```